dockerfiles/init/modules/compose/templates/docker-compose.yml.erb

```diff
@@ -145,7 +152,7 @@ services:
       - postgres:postgres
     volumes:
       - '<%= scope.lookupvar('che::che_instance') -%>/config/keycloak/che:/opt/jboss/keycloak/themes/che'
-      - '<%= scope.lookupvar('che::che_instance') -%>/config/keycloak/:/export'
```

Comment:
does this change a way how realms are added? because AFAIK keycloak use /export folder to export configs which work only once. if we mount our files to `/opt/jboss/keycloak/realms/` does this will change that behavior? I mean each further change will also applied not as it was with `/export`

Reply:
No, it doesn't since overriding previous realms is driven by the following parameter:

    -Dkeycloak.migration.strategy=IGNORE_EXISTING

The first time I deploy the Keycloak server (with an empty persistent volume), I see these messages in the start logs:
```
11:15:19,806 INFO  [org.keycloak.exportimport.dir.DirImportProvider] (ServerService Thread Pool -- 50) Importing from directory /opt/jboss/keycloak/realms
--
  | 11:15:19,819 INFO  [org.keycloak.services] (ServerService Thread Pool -- 50) KC-SERVICES0030: Full model import requested. Strategy: IGNORE_EXISTING
  | 11:15:23,036 INFO  [org.keycloak.exportimport.util.ImportUtils] (ServerService Thread Pool -- 50) Realm 'master' imported
  | 11:15:23,910 INFO  [org.keycloak.exportimport.dir.DirImportProvider] (ServerService Thread Pool -- 50) Imported users from /opt/jboss/keycloak/realms/master-users-0.json
  | 11:15:25,807 INFO  [org.keycloak.exportimport.util.ImportUtils] (ServerService Thread Pool -- 50) Realm 'che' imported
  | 11:15:26,070 INFO  [org.keycloak.exportimport.dir.DirImportProvider] (ServerService Thread Pool -- 50) Imported users from /opt/jboss/keycloak/realms/che-users-0.json
  | 11:15:26,146 INFO  [org.keycloak.services] (ServerService Thread Pool -- 50) KC-SERVICES0032: Import finished successfully
```

But if I restart the keycloak server (with its data already available in the persistent volume), I see the following message in the logs: 
```
11:36:15,183 INFO  [org.keycloak.exportimport.dir.DirImportProvider] (ServerService Thread Pool -- 60) Importing from directory /opt/jboss/keycloak/realms
--
  | 11:36:16,264 INFO  [org.keycloak.services] (ServerService Thread Pool -- 60) KC-SERVICES0030: Full model import requested. Strategy: IGNORE_EXISTING
  | 11:36:16,744 INFO  [org.keycloak.exportimport.util.ImportUtils] (ServerService Thread Pool -- 60) Realm 'master' already exists. Import skipped
  | 11:36:16,950 INFO  [org.keycloak.exportimport.util.ImportUtils] (ServerService Thread Pool -- 60) Realm 'che' already exists. Import skipped
  | 11:36:16,998 INFO  [org.keycloak.services] (ServerService Thread Pool -- 60) KC-SERVICES0032: Import finished successfully
```

And I can confirm that the settings done in the Keycloak adm console are kept after restart.